### DISCUSSION TITLE
honeygain failed startup

### DISCUSSION THREAD

**Discussion by robihut** [engageub/InternetIncome#157] [Q&A]
please help error during boot
what is the solution please help
thanks a lot
![screenshot_1728225771](https://github.com/user-attachments/assets/76a237b8-8c95-4177-bfe2-dcac0d4b26b2)
![screenshot_1728225830](https://github.com/user-attachments/assets/cc03e045-4065-4229-a714-bdcbb2a53df9)

**Comment by engageub** [comment]
Hello,
It looks like you have not added proxies in correct format or tun2socks did not start.
You can increase the delay in properties file if your CPU is slow. 


Thank you

**Reply by robihut** [reply]
it works. i added wrong proxy
thanks a lot

**Comment by engageub** [comment]
If you are using arm64 architecture, please remove `--platform=linux/amd64` from `honeygain_image`.

Thank you

**Reply by engageub** [reply]
Main branch does not have option to set a delay, the default value is 1 second.
Please use [test](https://github.com/engageub/InternetIncome/tree/test) branch and increase  `DELAY_BETWEEN_CONTAINER` in properties.conf if your PC is slow.

Thank you

**Reply by engageub** [reply]
If you wish to use the same code and update the delay, please increase the delay in sleep command as mentioned in the following screenshot. 
Please confirm if that resolved the issue. Main branch also will be provided with delay option if that resolved the issue.

![image](https://github.com/user-attachments/assets/84d2100a-d7e5-43bb-8e5b-c2113b1091d6)


Thank you

**Comment by robihut** [comment]
by the way i want to ask your code has fake Device Info ?
i can run multiple accounts each account 1 Device Info is that possible ?
please guide me
thanks a lot

**Reply by engageub** [reply]
Hello,
The code does not use fake Device Info.
Docker containers run in isolated environment which already have OS defined.
You can limit the CPU and memory for each container in test branch through properties file. 
The main purpose of these applications is to redirect the traffic and few applications have options to add proxies directly in their dashboard instead of running the applications. 
Although it is possible to change the device info by overriding the system files, it depends on what OS is being used inside the container and what commands the application uses to get the device info.

Thank you

**Reply by mrphuong186** [reply]
any way to override system info please advise
thank you very much

**Reply by engageub** [reply]
Hello,

There was a discussion regarding memory limits in Docker containers. Certain commands can operate effectively within the confines of Docker’s memory limits. However, other commands must be overridden by mounting them as volumes prior to container initialization. For instance, commands such as `cpuinfo` and `meminfo` need to be overridden before starting the container, as they may vary depending on the operating system.

It is important to note that `meminfo` may require frequent updates, as its values can fluctuate in real-time based on the applications running. Additionally, random memory values for disk free space need to be provided at one-minute intervals, or the values should be obtained from the cgroup memory command. While this task is not insurmountable, as it is ultimately software-based and the application’s behaviour is contingent upon the data it receives, it does require careful management.

There are complexities to consider, particularly if the application behaves differently based on the underlying OS architecture.

https://forums.docker.com/t/docker-memory-limit-does-not-show-inside-the-container/141503

Thank you

**Reply by mrphuong186** [reply]
thanks for your support
it seems the problem you mentioned is technical issue with little knowledge of linux, i need more detailed instructions.
specifically i use ubutu from vmware, how to spoof device info for each proxy?
it seems many sites don't allow ubutu from vmware to work because of device info for example honeygain
you can message here or email: maketview@gmail.com
i will be very grateful to you.
![screenshot_1728372788](https://github.com/user-attachments/assets/4db88797-8a2e-431c-ac03-45dd14c6d7d1)

**Reply by engageub** [reply]
Hello,
It appears that the issue may not be related to device information but rather your IP address. If you are not seeing any traffic, it's possible that your subnet might be blocked. I recommend checking the logs of the Honeygain Docker container for further insights.

Please ensure that you are using the latest code from GitHub, as recent updates have been made concerning DNS functionality.

In my recent experience with ten devices, I did not encounter any issues while utilizing multiple IP addresses. However, modifying device information can be complex, as it depends on the application’s design. If the Honeygain Docker application is configured to use only Docker as a prefix, you will need to download the Linux application and rebuild the Docker image accordingly. The suggestions provided in the previous comment are applicable, assuming you possess a working knowledge of Docker and Linux.

It is important to note that Honeygain identifies containers by their device names. If you use the default device name on two different hosts, both will register as `ubuntu1` when employing proxies, potentially leading to account blocks. Therefore, if you are running Honeygain Docker on multiple hosts, please ensure that the device names in the properties file are unique.


Thank you

**Reply by mrphuong186** [reply]
i reinstalled ubuntu from where and, reset modem, created new account on new device on new IP
and no traffic seen
How to fix blocked subnet?
please guide me
thanks a lot
![screenshot_1728387107](https://github.com/user-attachments/assets/5b945de6-9189-4b53-9a3e-3cf44dc9aab0)

**Reply by engageub** [reply]
If your IP is detected as non-residential or subnet is blocked by honeygain, we can't do anything about it. 
You can look into your docker container logs using the following command after enabling logs in `properties.conf` file to see if there are any issues.

```
sudo docker container logs <YOUR_HONEYGAIN_CONTAINER_NAME>
```

Sometimes, it may take about an hour to get the traffic.  

Thank you

**Reply by mrphuong186** [reply]
strange i use test version and it works
official version doesn't
very grateful to you

**Reply by engageub** [reply]
Hello,
The main branch and test branch use version 0.6.6 for amd64 architecture.
May be you were using older version of main branch.
Official Docker image of honeygain is being used in the code.

Thank you

**Reply by mrphuong186** [reply]
using version 0.6.6 can see green ip appear on dashboard, but point income does not increase
version 0.8.1 no connection on dashboard
i am using ubuntu version 22.04.5
for few days i can not fix it
thank you very much
![screenshot_1728429549](https://github.com/user-attachments/assets/808a3afe-2b5e-47ca-b530-c5255064da8f)
![screenshot_1728429686](https://github.com/user-attachments/assets/42c6c9d7-f5ee-45b9-90e8-0567e61be8c1)

**Comment by engageub** [comment]
Hello,
If it shows green and no traffic, it may be due to demand in location or IP is blacklisted. However, please look into your honeygain logs for any potential reasons that might be causing this.
From the screenshot mentioned above, your honeygain container name is present. 
Replace your honeygain container name in the following command.
Please make sure to enable logs in properties.conf file before running the command.

```
sudo docker container logs honey27d4f...........................e1
```

Thank you

**Reply by mrphuong186** [reply]
this is the result i got from the log
what is the problem ? please help
thanks a lot
![screenshot_1728475023](https://github.com/user-attachments/assets/e393fdd1-9193-4316-b83c-35d332bf374f)

**Reply by Raffy142** [reply]
Hi, u need change the ip, i have ur same problem and with change ip the problem is solved

**Reply by mrphuong186** [reply]
> Hi, u need change the ip, i have ur same problem and with change ip the problem is solved

thank you i will try it right away
